packages/app/src/state/clients/graphqlClient/index.ts

```diff
@@ -47,6 +47,14 @@ export const isGraphQLClientError = (error: Error): error is ClientError => {
   return error instanceof ClientError;
 };
 
+export const getGraphQLClientErrorMessage = (error: ClientError): string => {
+  try {
+    return error.response.errors![0].message;
+  } catch (e) {
+    return `GraphQL Error (Code: ${error.response.status})`;
```

Comment:
Is `error.response` guaranteed to not be `undefined`?

In other words: if there was a transient issue and the request was not sent out, or no response arrived -- are we sure we do not call this function?

Reply:
Yes - as `error` must be a `graphql-request` [`ClientError`](https://github.com/prisma-labs/graphql-request/blob/58a6d369d075b7b6cbd5ed0d6b1917c35786fe03/src/types.ts#L25-L55) (see [parameters](https://github.com/opstrace/opstrace/pull/1209/files#diff-129b93e0d8c903400c4b13ba4324668831194e24a3bab17df813dcdda1368eb0R50)).

`ClientError.response` [must not be undefined](https://github.com/prisma-labs/graphql-request/blob/20f18d3be52c85b4a0928ed8fe9ebba736e26f4c/src/types.ts#L26) and [is provided by `graphql-request`](https://github.com/prisma-labs/graphql-request/blob/20f18d3be52c85b4a0928ed8fe9ebba736e26f4c/src/index.ts#L229).